src/test/kotlin/uk/gov/dluhc/printapi/service/PrintResponseProcessingServiceTest.kt

```diff
@@ -166,6 +174,80 @@ class PrintResponseProcessingServiceTest {
             assertThat(captor.allValues[1]).usingRecursiveComparison().isEqualTo(listOf(expectedPrintDetails2))
             verify(idFactory).requestId()
         }
+
+        @Test
+        fun `should update all certificates within each batch`() {
+            // Given
+            val batchResponse1 = buildBatchResponse(status = FAILED)
+            val batchResponse2 = buildBatchResponse(status = SUCCESS)
+            val batchId1 = batchResponse1.batchId
+            val batchId2 = batchResponse2.batchId
+            val certificate1 = certificateBuilder(
+                printRequests = listOf(
+                    printRequestBuilder(
+                        batchId = batchId1,
+                        printRequestStatuses = listOf(
+                            printRequestStatusBuilder(
+                                status = SENT_TO_PRINT_PROVIDER,
+                                eventDateTime = batchResponse1.timestamp.toInstant().minusSeconds(10)
+                            )
+                        )
+                    )
+                )
+            )
+            val certificate2 = certificateBuilder(
+                printRequests = listOf(
+                    printRequestBuilder(
+                        batchId = batchId2,
+                        printRequestStatuses = listOf(
+                            printRequestStatusBuilder(
+                                status = SENT_TO_PRINT_PROVIDER,
+                                eventDateTime = batchResponse2.timestamp.toInstant().minusSeconds(10)
+                            )
+                        )
+                    )
+                )
+            )
+
+            val newRequestId = aValidRequestId()
+            given(certificateRepository.findByStatusAndPrintRequestsBatchId(any(), any()))
+                .willReturn(listOf(certificate1), listOf(certificate2))
+            given(idFactory.requestId()).willReturn(newRequestId)
+
+            // When
+            service.processBatchResponses(listOf(batchResponse1, batchResponse2))
+
+            // Then
+            verify(certificateRepository).findByStatusAndPrintRequestsBatchId(SENT_TO_PRINT_PROVIDER, batchResponse1.batchId)
+            verify(certificateRepository).findByStatusAndPrintRequestsBatchId(SENT_TO_PRINT_PROVIDER, batchResponse2.batchId)
+            verify(certificateRepository).saveAll(listOf(certificate1))
+            verify(certificateRepository).saveAll(listOf(certificate2))
+            assertThat(certificate1.getCurrentPrintRequest().requestId).isEqualTo(newRequestId)
+            assertThat(certificate1.getCurrentPrintRequest().batchId).isNull()
+            assertThat(
```

Comment:
As this is not a `data` class no option to use the `copy` approach used in the previous DynamoDb tests.